docs/content/guide/toml.md

```diff
@@ -82,3 +82,29 @@ This should make it pretty straightforward to make changes to lots of nodes with
 ## TOML groups
 
 They work exactly the same way as nodes, but have their own `groups/` directory. `.toml`, `.toml_set()` and `toml_save()` are also found on `Group` objects.
+
+<br>
+
+## Using and managing secrets
+
+Due to TOML nodes not supporting python code, BundleWrap supports what we call "magic strings".
+These allow you to use secrets in your nodes with just using a special syntax inside a string.
+
+To define magic strings, you need to add a `magic-strings.py` to your repository. It might look like
+this example:
+
+```python
+@magic_string
+def decrypt(string):
+    return vault.decrypt(string)
+```
+
+In your node you then use the function as follows:
+
+```toml
+mysecret = "!decrypt:SecretStringEncryptedBy `bw pw -e`"
+```
+
+The part between `!` and `:` is used as the function name, everything after the `:` will be passed
+as argument to the called function. Bundlewrap will raise `InvalidMagicStringException` if the
+function cannot be found.
```

Comment:
Is there a recommended way to use this new mechanism to declare stuff as `atomic()` in TOML nodes? If so, should we include this in the docs?

This is what I came up with now:

```python
from bundlewrap.metadata import atomic as _bw_atomic


@magic_string
def atomic(arg):
    return _bw_atomic(eval(arg))
```
The `eval()` isn’t particularly pretty, though. 🤔